org.eclipse.jdt.core.manipulation/proposals/org/eclipse/jdt/internal/ui/text/correction/SuppressWarningsSubProcessorCore.java

```diff
@@ -38,4 +64,133 @@ public static final boolean hasSuppressWarningsProposal(IJavaProject javaProject
 		return false;
 	}
 
+	public static void addSuppressWarningsProposals(IInvocationContextCore context, IProblemLocationCore problem, Collection<ChangeCorrectionProposalCore> proposals) {
```

Comment:
It looks to me like you've taken https://github.com/eclipse-jdt/eclipse.jdt.ui/blob/8e28ae0210672ac240683b778206bed85263fa66/org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/text/correction/SuppressWarningsSubProcessor.java#L67 and copied it over here. To have it be a "move" you'd need to remove the method and its helpers from that location. Then, everywhere in the codebase that called that method, would get replaced with the new location here. (eg. `SuppressWarningsSubProcessor.addSuppressWarningsProposals` -> `SuppressWarningsSubProcessorCore.addSuppressWarningsProposals`.

Also, what about `addUnknownSuppressWarningProposals` & `addRemoveUnusedSuppressWarningProposals` ?

Reply:
I'll update that.

I didn't move `addUnknownSuppressWarningProposals` & `addRemoveUnusedSuppressWarningProposals` because they weren't needed for the user request in https://github.com/eclipse-jdtls/eclipse.jdt.ls/issues/2698. I can certainly move these as well to be used for additional suggestions in quickfix.

Reply:
Fair. If it's not too much extra effort to move them as well, I would go for it.